### PR TITLE
New Object management endpoint added

### PR DESCRIPTION
**Pull request by danielzumb**

### CODE REVIEW

File: lib/rest.js [hunk 1]
```diff
@@ -27,6 +28,9 @@ module.exports = class MicroStrategyRESTAPI extends RestConnection {
     // https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Reports
     this.reports = new ReportsAPI(this);
 
+    // https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Object%20Management
+    this.objectManagement = new ObjectManagementAPI(this);
```
Comment (tiagosiebler) [inline lib/rest.js]: I missed the mark here in the user management PR, but I meant to keep alphabetical order here (same as the api-docs page). Can you move objectManagement and userManagement to the right positions? It's minor now but can quickly get messy while we add the other endpoints.

File: lib/restAPIs/objectManagement.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  recommendObject(objectId, offset, limit, fields){
+    assert(objectId, 'No objectId provided');
+
+    const queryParameter = {
+      offset: offset,
+      limit: limit
+    }
+    if (fields) {
+      queryParameter.fields = fields;
```
Comment (tiagosiebler) [inline lib/restAPIs/objectManagement.js]: I think it's worth waiting for #9 to merge then leveraging that fn for this trio of repeated parameters.

File: samples/objectManagement/modifyObjects.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  // MicroStrategy Tutorial
+  const projectId = 'B19DEDCC11D4E0EFC000EB9495D0F44F';
+  mstrApi.setProjectId(projectId);
+
+  const ObjectManagementAPI = mstrApi.objectManagement;
+
+  const objectID = '6DBBD2404562BC0C580E1B8B18DAEAA8';
+  const type = 8; // Integer from EnumDSSXMLObjectTypes
+
+  const copyObjecctBody = {
```
Comment (tiagosiebler) [inline samples/objectManagement/modifyObjects.js]: Minor but typo in var name.

**Review by tiagosiebler — CHANGES_REQUESTED** [review]
Looks good, just some minor remarks.

File: lib/restAPIs/objectManagement.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    const queryParameter = {
+      type: type,
+      certify: certify
+    };
+
+    return this._makeRequest(`${this._getBase()}/${objectId}/certify`, {}, 'PUT', this.getProjectHeader(), queryParameter)
+    .then(result => this.throwIfFailed(result, 200));
+  }
+
+  recommendObject(objectId, offset, limit, fields){
```
Comment (tiagosiebler) [inline lib/restAPIs/objectManagement.js]: I think this should be `getRecommendations`

Comment (tiagosiebler) [reply lib/restAPIs/objectManagement.js]: https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Object%20Management/getRecommendations

Comment (danielzumb) [reply lib/restAPIs/objectManagement.js]: The curl URL when you test it on the page is:

`https://demo.microstrategy.com/MicroStrategyLibrary/api/objects/12345/**recommendations**?offset=0&limit=-1`

which just uses recommendations. Also threw it in a sample to test and it gives me recommendations back for a dossier so I think it's all good

**Review by danielzumb — COMMENTED** [review]
The getRecommendations endpoint URL is actually just /recommendations/. Tested it in a sample and check the curl of the docs page

File: lib/restAPIs/objectManagement.js (hunk 4 above)
Comment (tiagosiebler) [reply lib/restAPIs/objectManagement.js]: The prefix denotes the purpose. This API doesn't set an object as recommended, it GETs a list of recommendations based on the query. From my perspective "recommend" seems a bit like "setRecommended" - thought "getRecommendations" would be more intuitive & self documenting from the fn name alone.

Comment (danielzumb) [reply lib/restAPIs/objectManagement.js]: Ah I see what you were saying. Thought you meant the API endpoint, yeah that name makes sense, i've updated it

**Review by tiagosiebler — APPROVED** [review]
👍

**Pull request merged**